x/oracle/keeper/keeper.go

```diff
@@ -280,6 +280,17 @@ func (k Keeper) GetAggregateExchangeRatePrevote(
 	return aggregatePrevote, nil
 }
 
+// HasAggregateExchangeRatePrevote checks if a validator has an existing prevote.
+func (k Keeper) HasAggregateExchangeRatePrevote(
+	ctx sdk.Context,
+	voter sdk.ValAddress,
+) bool {
+	if _, err := k.GetAggregateExchangeRatePrevote(ctx, voter); err != nil {
```

Comment:
You can use the `Has` method on the store btw. This is actually accrues less gas for the user.

Reply:
🧠